homeassistant/components/steamist/entity.py

```diff
@@ -24,13 +26,11 @@ def __init__(
         """Initialize the entity."""
         super().__init__(coordinator)
         self.entity_description = description
-        if coordinator.device_name:
-            self._attr_name = f"{coordinator.device_name} {description.name}"
         self._attr_unique_id = f"{entry.entry_id}_{description.key}"
         if entry.unique_id:  # Only present if UDP broadcast works
             self._attr_device_info = DeviceInfo(
                 connections={(dr.CONNECTION_NETWORK_MAC, entry.unique_id)},
-                name=entry.data[CONF_NAME],
+                name=coordinator.device_name,
```

Comment:
```suggestion
                name=entry.data[CONF_NAME],
```

If you want to do this, the name needs to be set after discovery in `__init__.py` `async_setup_entry`

Reply:
How do you mean exactly? The line adding the name to the device says `entry.data.get(CONF_NAME),  # Only found from discovery` so that indicates that the name can be None. So wouldn't that make entry.data[CONF_NAME] breaking?

Reply:
The block is guarded with `if entry.unique_id` so the name won't be none if the unique id is set

Reply:
But when unique_id is none, there is no device so the naming would be messed up right?

Reply:
Sorry it took me so long to get back to this.  I think its fine either way since the `if entry.unique_id` will prevent it from being a problem